swagger2.0/oic.r.rule.swagger.json

```diff
@@ -0,0 +1,371 @@
+          }
+        }
+      },
+      "post": {
+        "description": "",
+        "parameters": [
+          {"$ref": "#/parameters/interface-create"},
+          {
+            "name": "body",
+            "in": "body",
+            "required": true,
+            "schema": { "$ref": "#/definitions/rule-create" },
+            "x-example":
+              {
+                "rt": ["oic.r.rule.expression"],
+                "if": ["oic.if.rw","oic.if.baseline"],
+                "p":  {"bm": 3},
+                "rep": {
+                  "rule": "(timeperiod:triggered = true)",
+                  "ruleresult": false,
+                  "ruleenable": false,
+                  "actionenable": false
+                }
+              }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description" : "",
+            "x-example": {
+              "href": "/newURIforResource",
+              "ins": 39724818,
+              "rt": ["oic.r.rule.expression"],
+              "if": ["oic.if.rw","oic.if.baseline"],
+              "p":  {"bm": 3},
+              "rep": {
+                "rule": "(timeperiod:triggered = true)",
+                "ruleresult": false,
+                "ruleenable": false,
+                "actionenable": false
+              }
+            },
+            "schema": { "$ref": "#/definitions/rule-create" }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "interface-all": {
+      "in": "query",
+      "name": "if",
+      "type": "string",
+      "enum": ["oic.if.ll","oic.if.baseline","oic.if.create"]
+    }
+  },
+  "definitions": {
+    "rule-create": {
+      "type": "object",
+      "properties": {
+        "href": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.links.properties.core-schema.json#/definitions/href"
+        },
+        "if": {
+          "description": "The OCF Interfaces supported by the created Resource",
+          "items": {
+            "enum": [
+              "oic.if.baseline",
+              "oic.if.ll",
+              "oic.if.b",
+              "oic.if.rw",
+              "oic.if.r",
+              "oic.if.a",
+              "oic.if.s"
+            ],
+            "type": "string",
+            "maxLength": 64
+          },
+          "minItems": 1,
+          "uniqueItems": true,
+          "type": "array"
+        },
+        "ins": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.links.properties.core-schema.json#/definitions/ins"
+        },
+        "p": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.links.properties.core-schema.json#/definitions/p"
+        },
+        "rt": {
+          "description": "Resource Type of the created Resource",
+          "items": {
+            "maxLength": 64,
+            "type": "string",
+            "enum": ["oic.r.rule.inputcollection","oic.r.rule.expression","oic.r.rule.actioncollection"]
```

Comment:
Why are all of these in the "rule-create" definition?  Is it possible to actually CREATE a Rule Input Collection and Rule Action Collection?  If so, I think this all needs to be discussed in the CR.  Setting up an oic.r.rule Resource could be very complicated if all of these Resources need to be created.

Reply:
The CR does discuss this.  The basic expectation is that they're already present.  But they can be created.  If this is too complex for day #1 I can remove this.

Reply:
Yes, sorry. I was commenting on each schema right after I read the section that defined it while the section was still fresh in my mind.  This is explained near the end of the CR in clause 11.10.3 Rule Behaviour.  This comment was added before I had read that section.
How do we mark this conversation as resolved?